src/oop/special_class_method/medium/q10/MainTest.java

```diff
@@ -0,0 +1,275 @@
+package oop.special_class_method.medium.q10;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private final String className = "Shoe";
+    private final String attributeName1 = "size";
+    private final String getAttributeMethodName1 = "getSize";
+    private final String setAttributeMethodName1 = "setSize";
+    private final String attributeName2 = "colour";
+    private final String getAttributeMethodName2 = "getColour";
+    private final String setAttributeMethodName2 = "setColour";
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() throws Throwable {
+        String classString = "oop.special_class_method.medium.q10." + className;
+        this.testClass = new ObjectTest(classString);
+        String modifiedClassMessage =
+                "You have modified the provided %s attribute. Please revert it to the original state.";
+        assertTrue(testClass.hasField(attributeName1, int.class, new String[]{"private"}),
+                String.format(modifiedClassMessage, attributeName1));
+        assertTrue(testClass.hasField(attributeName2, String.class, new String[]{"private"}),
+                String.format(modifiedClassMessage, attributeName2));
+        Object[][] tests = new Object[][]{
+                {3, "Red"},
+                {10, "Black"},
+                {12, "Blue"},
+                {43, "White"},
+                {7, "Yellow"}
```

Comment:
add in a 0 test and a below 0 test to ensure they didn't add an if statement or something